packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2867,8 +2940,33 @@ class InputDecoration {
   /// {@endtemplate}
   ///
   /// If null, [InputDecorationTheme.floatingLabelBehavior] will be used.
+  ///
+  /// See also:
+  ///
+  ///  * [floatingLabelAlignment] which defines where the floating label should
+  ///    be displayed.
   final FloatingLabelBehavior? floatingLabelBehavior;
 
+  /// {@template flutter.material.inputDecoration.floatingLabelAlignment}
+  /// Defines **where** the floating label should be displayed.
+  ///
+  /// [FloatingLabelAlignment.start] aligns the floating label to the leftmost
+  /// (when [TextDirection.ltr]) or rightmost (when [TextDirection.rtl]),
+  /// possible position, which is vertically adjacent to the label, on top of
+  /// the field.
+  ///
+  /// [FloatingLabelAlignment.center] aligns the floating label to the center on
+  /// top of the field.
+  /// {@endtemplate}
+  ///
+  /// If null, [InputDecorationTheme.floatingLabelAlignment] will be used.
+  ///
```

Comment:
Nit: Maybe add one more paragraph to clarify:

```
When the label is not floating, then floatingLabelAlignment has no effect.
```

Is that too obvious?  I'm just worried someone might toggle this parameter and be confused when nothing happens because floatingLabelBehavior is `never` or something.

We could also add an assertion in the constructor that floatingLabelBehavior can't be `never` when floatingLabelAlignment is set...  That seems like overdoing it though.

Reply:
IMO having **floating** in its name sufficiently indicates that it'll affect label only when it's floating.

I don't think floatingLabelAlignment has a set/unset state as it can be either of two values, start or center. So an assertion may not help.

Reply:
Sounds good, I'm fine keeping this as-is.